infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationStreamNameTableManager.java

```diff
@@ -258,6 +240,58 @@ private Set<String> readStreamsToReplicateFromTable() {
         return tableNames;
     }
 
+    /**
+     * Helper method for checking whether LR is in upgrading path or not.
+     * Note that the default boolean value for ProtoBuf message is false.
+     *
+     * @return True if LR is in upgrading path, false otherwise.
+     */
+    public boolean isUpgraded() {
+        VersionString versionString = VersionString.newBuilder()
+                .setName(VERSION_PLUGIN_KEY).build();
+        CorfuStoreEntry<VersionString, Version, CommonTypes.Uuid> record;
+        try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+            record = txn.getRecord(LOG_REPLICATION_PLUGIN_VERSION_TABLE, versionString);
+            txn.commit();
+        } catch (NoSuchElementException e) {
```

Comment:
getRecord will only return a null if version table is not initialized.  Can you change this condition accordingly?

Reply:
Here the `NoSuchElementException` is for guarding the case that this method is accessed before version table initiated.